### ISSUE TITLE
"Remove redundant parentheses" false positive

### ISSUE THREAD

**Issue by benwh1** [rust-lang/rust-analyzer#17111]
**rust-analyzer version**: v0.3.1924

**rustc version**: rustc 1.79.0-nightly (9d5cdf75a 2024-04-07)

**editor or extension**: VSCode

**code snippet to reproduce**: 
```rust
struct Foo<T> {
    t: T,
}

impl Foo<fn(u32)> {
    fn foo(&self) {
        (self.t)(42);
    }
}
```

The parentheses around `self.t` are required but rust-analyzer suggests the "remove rendundant parentheses" action.
